runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -119,20 +124,32 @@ public Object getPayload(CorfuRuntime runtime) {
     @Override
     public synchronized void releaseBuffer() {
         if (serializedCache != null) {
-            serializedCache.release();
-            if (serializedCache.refCnt() == 0) {
+            serializedCache.buffer.release();
+            if (serializedCache.buffer.refCnt() == 0) {
                 serializedCache = null;
             }
         }
     }
 
     @Override
-    public synchronized void acquireBuffer() {
-        if (serializedCache == null) {
-            serializedCache = Unpooled.buffer();
-            doSerializeInternal(serializedCache);
+    public synchronized void acquireBuffer(boolean metadata) {
+        if (serializedCache != null) {
```

Comment:
if you invert the `if` condition it makes code a bit more straightforward
```
    public synchronized void acquireBuffer(boolean metadata) {
        if (serializedCache == null) {
            ByteBuf buf = Unpooled.buffer();
            int metadataOffset;
            
            if (metadata) {
                metadataOffset = doSerializeInternal(buf);
            } else {
                doSerializePayloadInternal(buf);
                metadataOffset = buf.writerIndex();
            }
            serializedCache = new SerializedCache(buf, metadataOffset);
        } else {
            if (metadata) {
                serializedCache.buffer.resetReaderIndex();
                serializedCache.buffer.writerIndex(serializedCache.metadataOffset);
                doSerializeMetadataInternal(serializedCache.buffer);
            }
            serializedCache.buffer.retain();
        }
    }
```

Reply:
Done.